Composer/packages/client/src/pages/notifications/index.tsx

```diff
@@ -3,35 +3,41 @@
 
 /** @jsx jsx */
 import { jsx } from '@emotion/core';
-import { useState } from 'react';
+import { useState, useContext } from 'react';
 import { RouteComponentProps } from '@reach/router';
-
+import { StoreContext } from '../../store';
 import { ToolBar } from './../../components/ToolBar/index';
 import useNotifications from './useNotifications';
 import { NotificationList } from './NotificationList';
 import { NotificationHeader } from './NotificationHeader';
 import { root } from './styles';
 import { INotification } from './types';
 import { navigateTo } from './../../utils';
-import { convertDialogDiagnosticToUrl } from './../../utils/navigation';
+import { convertDialogDiagnosticToUrl, convertInlineLgDiagnosticToUrl } from './../../utils/navigation';
 
-const navigations = {
-  lg: (item: INotification) => {
-    navigateTo(`/language-generation/${item.id}/edit#L=${item.diagnostic.range?.start.line || 0}`);
-  },
-  lu: (item: INotification) => {
-    navigateTo(`/dialogs/${item.id}`);
-  },
-  dialog: (item: INotification) => {
-    //path is like main.trigers[0].actions[0]
-    //uri = id?selected=triggers[0]&focused=triggers[0].actions[0]
-    const uri = convertDialogDiagnosticToUrl(item.diagnostic);
-    navigateTo(uri);
-  },
-};
 const Notifications: React.FC<RouteComponentProps> = () => {
   const [filter, setFilter] = useState('');
+  const { state } = useContext(StoreContext);
+  const { dialogs } = state;
   const notifications = useNotifications(filter);
+  const navigations = {
+    inlineLgTemplate: (item: INotification) => {
```

Comment:
I prefer keep consistent `lg` category, distinguish inline or not with INotification.id, case INotification.type here most means file(source) type.. for example use naming `fileId#templateId`  for inline , `fileId` for non-inline.